samples/spring-transfer-workflow/src/main/java/com/example/transfer/TransferWorkflowEntity.java

```diff
@@ -0,0 +1,181 @@
+package com.example.transfer;
+
+import com.example.transfer.FraudDetectionResult.TransferRejected;
+import com.example.transfer.FraudDetectionResult.TransferRequiresManualAcceptation;
+import com.example.transfer.FraudDetectionResult.TransferVerified;
+import com.example.transfer.TransferState.Transfer;
+import kalix.javasdk.workflowentity.WorkflowEntity;
+import kalix.springsdk.KalixClient;
+import kalix.springsdk.annotations.EntityKey;
+import kalix.springsdk.annotations.EntityType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+import java.util.concurrent.CompletionStage;
+
+import static java.util.concurrent.CompletableFuture.completedFuture;
+
+// tag::class[]
+@EntityType("transfer") // <2>
+@EntityKey("transferId") // <3>
+@RequestMapping("/transfer/{transferId}") // <4>
+public class TransferWorkflowEntity extends WorkflowEntity<TransferState> { // <1>
+  // end::class[]
+
+  // tag::start[]
+  public record Withdraw(String from, int amount) { // <4>
+  }
+
+  // end::start[]
+
+  // tag::definition[]
+  public record Deposit(String to, int amount) { // <4>
+  }
+
+  // end::definition[]
+
+
+  private static final Logger logger = LoggerFactory.getLogger(TransferWorkflowEntity.class);
+
+  private final String fraudDetectionStepName = "fraud-detection";
+
+  // tag::definition[]
+  private final String withdrawStepName = "withdraw"; // <1>
+  private final String depositStepName = "deposit"; // <1>
+
+  // end::definition[]
+
+  final private KalixClient kalixClient;
+
+  public TransferWorkflowEntity(KalixClient kalixClient) {
+    this.kalixClient = kalixClient;
+  }
+
+  // tag::definition[]
+  @Override
+  public Workflow<TransferState> definition() {
+    Step withdraw =
+        step(withdrawStepName) // <1>
+            .call((Withdraw cmd) -> kalixClient.patch("/wallet/" + cmd.from() + "/withdraw/" + cmd.amount(), String.class)) // <2>
+            .andThen(this::moveToDeposit);
```

Comment:
Maybe better to use lambdas to make it self-contained. 

Also, the callouts numbers will be in order.